include/faabric/scheduler/MpiWorld.h

```diff
@@ -9,7 +9,8 @@
 #include <thread>
 
 namespace faabric::scheduler {
-typedef faabric::util::Queue<faabric::MPIMessage> InMemoryMpiQueue;
+typedef faabric::util::Queue<std::shared_ptr<faabric::MPIMessage>>
```

Comment:
Originally these were `std::unique_ptr` (which I preferred). This works generally allright except when trying to `peek` the queue. In that case we need a reference from the first element in the queue, but want to keep another reference in the queue. This particular use-case only made it necessary to switch to `shared_ptr`.